app/code/core/Mage/Eav/Model/Attribute/Data/Date.php

```diff
@@ -30,7 +30,11 @@ class Mage_Eav_Model_Attribute_Data_Date extends Mage_Eav_Model_Attribute_Data_A
     public function extractValue(Zend_Controller_Request_Http $request)
     {
         $value = $this->_getRequestValue($request);
-        return $this->_applyInputFilter($value);
+        $value = $this->_applyInputFilter($value);
+        if ($value === false) {
+            return '';
+        }
+        return $value;
```

Comment:
To preserve the original return type and to fix `Fatal error: Uncaught TypeError: Unsupported operand types: string - string`
```suggestion
        return empty($value) ? false : $this->_applyInputFilter($value);
````

Reply:
mmmm but in this case if $value = 0 it returns false, it shouldn't happen since it's a date but wouldn't it be better to use another check instead?

Reply:
`$value` is a `mixed` type. But I suspect it is `string|false`. It is a user input value in a form. So, if `$value == 0`, then user input is invalid, meaning, returning `false` is correct.

Reply:
mmm ok, I can't directly commit your code because of this error:

<img width="582" alt="Screenshot 2023-05-23 alle 22 27 40" src="https://github.com/OpenMage/magento-lts/assets/909743/49d2c74c-b010-4580-9d72-5da26f714f99">

I'll commit it from my IDE